### PR TITLE
[master] Porting #52933 to master

### PR DESCRIPTION
**Pull request by garethgreenaway**
Porting #52933 to master

### CODE REVIEW

**Review by dwoz — APPROVED** [review]

File: salt/beacons/inotify.py [hunk 1]
```diff
@@ -117,6 +117,10 @@ def validate(config):
         if 'files' not in _config:
             return False, 'Configuration for inotify beacon must include files.'
         else:
+            if not isinstance(_config['files'], dict):
```
Comment (s0undt3ch) [inline salt/beacons/inotify.py]: I would remove the above `else` and dedent this logic.

File: tests/unit/beacons/test_inotify.py [hunk 2]
```diff
@@ -39,10 +39,33 @@ def setUp(self):
     def tearDown(self):
         shutil.rmtree(self.tmpdir, ignore_errors=True)
 
+    def test_non_list_config(self):
+        config = {}
```
Comment (s0undt3ch) [inline tests/unit/beacons/test_inotify.py]: Unnecessary variable, just pass `{}` to `inotify.validate()`.

File: tests/unit/beacons/test_inotify.py [hunk 3]
```diff
@@ -39,10 +39,33 @@ def setUp(self):
...
+
+        ret = inotify.validate(config)
+
+        self.assertEqual(ret, (False, 'Configuration for inotify beacon must'
+                                      ' be a list.'))
+
     def test_empty_config(self):
         config = [{}]
```
Comment (s0undt3ch) [inline tests/unit/beacons/test_inotify.py]: Same as above.

File: tests/unit/beacons/test_inotify.py [hunk 4]
```diff
@@ -39,10 +39,33 @@ def setUp(self):
...
-        ret = inotify.beacon(config)
-        self.assertEqual(ret, [])
+        ret = inotify.validate(config)
+        _expected = (False, 'Configuration for inotify beacon must include files.')
+        self.assertEqual(ret, _expected)
+
+    def test_files_none_config(self):
+        config = [{'files': None}]
```
Comment (s0undt3ch) [inline tests/unit/beacons/test_inotify.py]: I won't say same as above here, although we don't actually use `config` anywhere else.
But for the sake of readability, let's leave it.
The same for the next test.

**Review by s0undt3ch — APPROVED** [review]
None of my review comments are blockers. It would just make the code better, more readable.

File: salt/beacons/inotify.py [hunk 5]
```diff
@@ -117,6 +117,10 @@ def validate(config):
...
+                return False, ('Configuration for inotify beacon invalid, '
+                               'files must be a dict.')
+
             for path in _config.get('files'):
 
                 if not isinstance(_config['files'][path], dict):
```
Comment (dhiltonp) [inline salt/beacons/inotify.py]: `Configuration for inotify beacon must be a list of dictionaries.`

This error is not correct, it must be a dict of dicts (which this PR validates).

**Review by dhiltonp — APPROVED** [review]

**Pull request merged**